src/lib/y2firewall/widgets/zones_table.rb

```diff
@@ -32,22 +32,74 @@ class ZonesTable < ::CWM::Table
       # Constructor
       #
       # @param zones [Array<Y2Firewall::Firewalld::Zone>] Zones
-      def initialize(zones)
+      # @param default_zone_button [Y2Firewall::Widgets::DefaultZoneButton] Button to change
+      #   the default zone
+      def initialize(zones, default_zone_button)
         textdomain "firewall"
         @zones = zones
+        @default_zone_button = default_zone_button
+      end
+
+      # @macro seeAbstractWidget
+      def opt
+        [:notify, :immediate]
+      end
+
+      # @macro seeAbstractWidget
+      def init
+        zone = Y2Firewall::UIState.instance.row_id
+        self.value = zone if zone && zones.map(&:name).include?(zone.to_s)
+        default_zone_button.zone = selected_zone
       end
 
       # @see CWM::Table#header
       def header
         [
           _("Name"),
-          _("Interfaces")
+          _("Interfaces"),
+          _("Default")
```

Comment:
I'm not sure about having an extra column for it, but probably that is something for Ken

Reply:
We have something similar in the repository list. But we might append a "(default)" string if preferred.